cli/command/stack/deploy_composefile.go

```diff
@@ -320,6 +322,13 @@ func deployServices(
 			if sendAuth {
 				updateOpts.EncodedRegistryAuth = encodedAuth
 			}
+
+			if image != service.Spec.TaskTemplate.ContainerSpec.Image {
```

Comment:
I think it would be best to just remove the check, so the digest and platform information are always updated.